lib/matplotlib/table.py

```diff
@@ -532,21 +567,34 @@ def _auto_set_font_size(self, renderer):
         if len(self._cells) == 0:
             return
         fontsize = next(iter(self._cells.values())).get_fontsize()
-        cells = []
+
+        grow = self._bbox is not None
+
         for key, cell in self._cells.items():
             # ignore auto-sized columns
             if key[1] in self._autoColumns:
                 continue
-            size = cell.auto_set_font_size(renderer)
-            fontsize = min(fontsize, size)
-            cells.append(cell)
 
-        # now set all fontsizes equal
-        for cell in self._cells.values():
+            # set initial guess at cell font size
             cell.set_fontsize(fontsize)
 
+            size = cell.auto_set_font_size(renderer, grow=grow)
+
+            # no point in trying bigger font after first
+            if grow:
+                grow = False
+                fontsize = size
+            else:
+                fontsize = min(fontsize, size)
+
+        # now set all fontsizes equal
+        self.set_fontsize(fontsize)
+
+        return fontsize
+
     def scale(self, xscale, yscale):
         """Scale column widths by *xscale* and row heights by *yscale*."""
+
```

Comment:
Unnecessary blank line introduced. Please remove.

Reply:
Blank line removed.

Not sure where to go from here.   There are other changes that I'd like to make to the code, specifically addressing https://github.com/matplotlib/matplotlib/issues/8830  -- I can create a second pull request with the attempt at fixing that issue, but it depends on the fixes here too, so was hoping to get this over the line before posting that.

Also have a pull request that allows cell edge colours as well as the cell face colour to be specified.

This is the second attempt I've made at fixing this stuff, I've probably spent close to 50-100 hours total on this and related issues and still no idea if/when this fix will be merged.